gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -470,7 +571,9 @@ private void submitJob(DagNode<JobExecutionPlan> dagNode) {
         // The SpecProducer implementations submit the job to the underlying executor and return when the submission is complete,
         // either successfully or unsuccessfully. To catch any exceptions in the job submission, the DagManagerThread
         // blocks (by calling Future#get()) until the submission is completed.
-        producer.addSpec(jobSpec).get();
+        Future addSpecFuture = producer.addSpec(jobSpec);
+        dagNode.getValue().setJobFuture(Optional.of(addSpecFuture));
```

Comment:
Also, don't you have to modify serialization/deserialization of Dag to add the future to the serialized dag in dagStateStore and deserialize it back on service restart?

Reply:
done

Reply:
Thanks!